docs/user_guide/ml.md

```diff
@@ -1,8 +1,254 @@
 # Machine Learning
 
-In this section, the users will learn how to implement machine learning in their analysis with
-columnflow.
+In this section, the users will learn how to implement machine learning in their analysis with columnflow.
 
-Note: which tasks, how to trigger them, how to train, test and plot networks and outputs
+# Configure your custom machine learning class:
+To create a custom machine learning (ML) class in columnflow, it is imperative to inherit from the MLModel class.
+This inheritance ensures the availability of functions to manage and access config and model instances, as well as the necessary producers.
+The name of your custom ML class can be arbitrary, since `law` accesses your machine learning model using a `cls_name` in {py::meth}`~columnflo.util.derive` e.g.
+```{literalinclude} ./ml_code.py
+:language: python
+:start-at: test_model
+:end-at: test_model
+```
+The second argument in {py::meth}`~columnflo.util.derive` is a `cls_dict` which configure your subclass.
+The `cls_dict` needs to be flat.
+The keys of the dictionary are set as class attributes and are therefore also accessible by using self.
+The configuration with derive has 2 main advantages:
+- manageability, since the dictionary can come from loading a config file and these can be changed fairly easy
+- flexibility, multiple settings require only different configuration files
```

Comment:
```suggestion
- ==manageability==, since the dictionary can come from loading a config file and these can be changed fairly easy
- ==flexibility==, multiple settings require only different configuration files
```

Reply:
OK I guess, but you didn't actually do this, correct? Why?

Reply:
I simply forgot to add this. To the batch, also = WORD = does not highlight in this markdown engine. I will just highlight with bold instead of HTML mark.
```suggestion
- **manageability**, since the dictionary can come from loading a config file and these can be changed fairly easy
- **flexibility**, multiple settings require only different configuration files
```